field.cpp

```diff
@@ -0,0 +1,194 @@
+#include "field.h"
+#include <random>
+#include <utility>
+
+Field::Field(int fieldSize){
+    setSize(fieldSize);
+    isMovable = true;
+}
+
+void Field::setSize(int fieldSize){
+    if(size != fieldSize){
+        size = fieldSize;
+
+        field.resize(size);
+        for (int i = 0; i < size; i++) {
+            field[i].resize(size);
+        }    size = fieldSize;
+    }
+}
+
+void Field::createField(int fieldSize){
+    setSize(fieldSize);
+    initializeField();
+}
+
+void Field::updateMovability(){
+    /*
+    * Field is movable if at least one cell is movable
+    */
+    for(int i = 0; i<size; i++){
+        for(int j = 0; j < size; j++){
+            //Cell *cell = field[i][j];
+            if(field[i][j].cellIsMovable()){
+                isMovable = true;
+                return;
+            }
+        }
+    }
+    isMovable = false;
+}
+
+void Field::initializeField(){
+    for(int i = 0; i < size; i++){
+        for(int j = 0; j < size; j++){
+            field[i][j] = Cell();
+
+            if(i-1>=0){ field[i][j].addNeighbor(Direction::UP, &field[i-1][j]); }
+            else{ field[i][j].addNeighbor(Direction::UP, nullptr); }
+
+            if(i+1 < size){ field[i][j].addNeighbor(Direction::DOWN, &field[i+1][j]); }
+            else{ field[i][j].addNeighbor(Direction::DOWN, nullptr); }
```

Comment:
You don't need those if/else here, from what I remember you already initiallized all the neighbors to nullptr when the cell had been created